lib/SILOptimizer/PassManager/PassPipeline.cpp

```diff
@@ -372,6 +372,7 @@ void addFunctionPasses(SILPassPipelinePlan &P,
   P.addAllocBoxToStack();
 
   if (P.getOptions().DestroyHoisting == DestroyHoistingOption::On) {
+    P.addComputeSideEffects();
```

Comment:
This pass is already invoked on other places in the pipeline. Is there a reason you added all this additional invocations?
Also note, that running the pass immediately before another pass (like destroy hoisting), doesn't immediately help the other pass, because compute-side-effects updates the summary for the _callees_ and not for the current function.

Reply:
Is there a reason you added all this additional invocations?
I didn't know if it was necessary or not and was hoping to hear whether it was.  Thanks!
> Also note, that running the pass immediately before another pass (like destroy hoisting), doesn't immediately help the other pass, because compute-side-effects updates the summary for the callees and not for the current function.
Yeah, the goal was to get summary info for all the function's _callees_ so that destroys/etc could be hoisted ever applies of those callees.

Reply:
Removed.